ODIN_II/SRC/odin_ii.cpp

```diff
@@ -183,14 +199,50 @@ static void optimization() {
     printf("\n--------------------------------------------------------------------\n");
 }
 
+/*---------------------------------------------------------------------------
+ * (function: get_current_dir)
+ returns current working directory
+ *-------------------------------------------------------------------------*/
+std::string get_current_dir() {
```

Comment:
similar routine already exist in odin_util
get_directory
if you need additional feature try to add a new utility function there